extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcProvider.java

```diff
@@ -503,6 +510,25 @@ public Key resolveKey(JsonWebSignature jws, List<JsonWebStructure> nestingContex
             }
         }
 
+        private Key findKeyByBruteForce(JsonWebSignature jws) {
```

Comment:
Not a bad idea but then even in the best case scenario, we'll run the signature verification twice.

I think we'll need to a higher level retry after a signature exception is thrown, inside verifyTokenInternal function. It can become a bit messy. I guess, if it is a `try-all` then a new instance of key resolver which remembers the already tried keys (or even the last index in the values collection) should be used.

Reply:
Adapted it, to check only the keys matching the key type.